problems/arraysWithLoops.js

```diff
@@ -24,40 +40,87 @@ function shoutWhileLoop() {}
  * @param {Array} - ["I", "am", "a", "happy", "array"]
  * @returns {Array} - ["I!", "am!", "a!", "happy!", "array!"]
  */
+    // const iterable = [10, 20, 30];
+
+// for (let value of iterable) {
+//   value += 1;
+//   console.log(value);
 
-function shoutForOfLoop() {}
+  const shoutForOfLoop = (array) => {
+    let newArray = []
+      for (let i of array) {
+        newArray.push(i + "!")  
+      }
+    return newArray 
+  }  
 
 /**
  * Takes in an array and returns the sum of all values
  * @param {number[]} nums
  * @returns {number} sum
  */
 
-function sumArray() {}
+ const sumArray = (nums) => {
+   let sum = 0;
+   for (let i = 0; i < nums.length ; i++) {
+      sum += nums[i] 
+   }
+   return sum 
+ }
 
 /**
  * Takes in an array and returns an array of all the odd valued elements
  * @param {number[]} nums
  * @returns {array} odds
  */
 
-function oddArray() {}
+// let numns = [ 1, 2, 3, 4, 5 ]
+
+const oddArray = (nums) =>{
+  let newArray = [] // newarray= [ 1,3  ] 
+  let i = 0;
+  while ( i < nums.length ){
+    if (nums [ i ] % 2 === 1){
+      newArray.push ( nums [ i ] )
+    }
+  i++                         
+  }
+  return newArray
+}
 
 /**
  * Takes in an array and returns an array of all the even valued elements
  * @param {number[]} nums
  * @returns {array} evens
  */
 
-function evenArray() {}
+const evenArray = (nums) => {
+  let newArray = [];
+  for(i = 0; i < nums.length; i++){
+    if(nums [ i ] % 2 === 0  ){
+      newArray.push ( nums [ i ] )  
+    }
+  }
+  return newArray;
+}
 
 /**
  * Takes in array and returns the smallest number in the array
  * @param {number[]} nums
  * @returns {number} smallest value
  */
 
-function findSmallest() {}
+const findSmallest = (number) =>{
+  let num = number [0];
+  for(i = 0; i < number.length; i++){
+    if( number[i] < num ){
+      num = number[i];
+    }else {
+      num = num;
```

Comment:
Here you are declaring the num variable to itself. No changes are being made in this code block.